src/components/PDFView/index.native.js

```diff
@@ -1,50 +1,133 @@
 
-const defaultProps = {
-    sourceURL: '',
-    style: {},
-};
+        // The message provided by react-native-pdf doesn't indicate whether this
+        // is an initial password request or if the password is invalid. So we just assume
+        // that if a password was already entered then it's an invalid password error.
+        if (this.state.password) {
+            this.setState({isPasswordInvalid: true});
+        }
+    }
 
-/**
- * On the native layer, we use a pdf library to display PDFs
- *
- * @param props
- * @returns {JSX.Element}
- */
+    /**
+     * When the password is submitted via PDFPasswordForm, save the password
+     * in state and attempt to load the PDF.
+     *
+     * @param {String} password Password submitted via PDFPasswordForm
+     */
+    attemptPdfLoadWithPassword(password) {
+        // Render PDF in invisible state. It's invisible because at this
+        // stage of the password challenge process isPdfLoaded is false.
+        this.setState({password, shouldAttemptPdfLoad: true});
+    }
+
+    /**
+     * When the PDF is successfully loaded, terminate password challenge by hiding
+     * PDFPasswordForm and making the PDF component visible.
+     */
+    terminatePasswordChallenge() {
+        // Don't render PDFPasswordForm, and do render PDF in visible state.
+        this.setState({shouldRequestPassword: false, isPdfLoaded: true});
+    }
+
+    render() {
+        const pdfStyles = [
+            styles.imageModalPDF,
+            StyleUtils.getWidthAndHeightStyle(this.props.windowWidth, this.props.windowHeight),
+        ];
+
+        // If we haven't yet successfully loaded the PDF, then we need to hide the
+        // react-native-pdf/PDF component so that PDFPasswordForm is positioned
+        // nicely. We're specifically hiding it because we still need to render the PDF so
+        // that it can validate the password.
+        if (!this.state.isPdfLoaded) {
+            pdfStyles.push(styles.invisible);
+        }
+
+        return (
+            <TouchableWithoutFeedback style={[styles.flex1, this.props.style]}>
+
+                <View>
+
+                    {this.state.shouldAttemptPdfLoad && (
+                        <PDF
+                            activityIndicator={<FullScreenLoadingIndicator />}
+                            source={{uri: this.props.sourceURL}}
+                            style={pdfStyles}
+                            onError={this.initiatePasswordChallenge}
+                            password={this.state.password}
+                            onLoadComplete={this.terminatePasswordChallenge}
+                        />
+                    )}
+
+                    {this.state.shouldRequestPassword && (
+                        <PDFPasswordForm
+                            onSubmit={this.attemptPdfLoadWithPassword}
+                            isPasswordInvalid={this.state.isPasswordInvalid}
```

Comment:
Same here remove unnecessary line breaks in between component usage in render method